app/components/search/publication-statuses-filter.js

```diff
@@ -0,0 +1,38 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { task } from 'ember-concurrency';
+
+export default class SearchPublicationStatusesFilterComponent extends Component {
+  @service store;
+
+  @tracked selectedStatusIds = this.args.selected || [];
+  @tracked selectedStatuses = [];
+  @tracked statuses = [];
+
+  constructor() {
+    super(...arguments);
+
+    this.loadPublicationStatuses.perform();
+  }
+
+  loadPublicationStatuses = task(async () => {
+    this.statuses = (await this.store.queryAll(
+      'publication-status',
+      { sort: 'position' },
+    )).toArray();
+
+    if (this.selectedStatusIds.length) {
+      this.selectedStatuses = this.statuses.filter(decisionResultCode =>
```

Comment:
**nitpick**: probably forgot to rename `decisionResultCode`. Not really important